Updated SharePoint Server link in connectors section

ebbypeter opened the pull request:
Both SharePoint Online and SharePoint Server links were pointing to 'connectors-create-api-sharepointonline'. Updated SharePoint Server link to point at 'connectors-create-api-sharepointserver'.

azurecla commented:
Hi __@v-kagri__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (v-kagri). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

ebbypeter committed:
Updated SharePoint Server link

Both SharePoint Online and SharePoint Server links were pointing to 'connectors-create-api-sharepointonline'. Updated SharePoint Server to connect to 'connectors-create-api-sharepointserver'.
